src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -714,6 +781,152 @@ AS
     VALUES
         (@studyKey, @seriesKey, @instanceKey, @studyInstanceUid, @seriesInstanceUid, @sopInstanceUid, @newWatermark, @initialStatus, @currentDate, @currentDate)
 
+    -- Insert Custom Tags
+    -- String Key tags
+    IF EXISTS (SELECT 1 FROM @stringCustomTags)
+    BEGIN      
+        MERGE INTO dbo.CustomTagString AS T
+        USING 
+        (
+            SELECT input.TagKey, input.TagValue, input.TagLevel 
+            FROM @stringCustomTags input
+            INNER JOIN dbo.CustomTag WITH (REPEATABLEREAD) 
+            ON dbo.CustomTag.TagKey = input.TagKey
+            AND dbo.CustomTag.TagStatus <> 2
+        ) AS S
+        ON T.TagKey = S.TagKey        
+            AND T.StudyKey = @studyKey
+            AND ISNULL(T.SeriesKey, @seriesKey) = @seriesKey 
+            AND ISNULL(T.InstanceKey, @instanceKey) = @instanceKey
+        WHEN MATCHED THEN 
+            UPDATE SET T.Watermark = @newWatermark, T.TagValue = S.TagValue
+        WHEN NOT MATCHED THEN 
+            INSERT (TagKey, TagValue, StudyKey, SeriesKey, InstanceKey, Watermark)
+            VALUES(
+            S.TagKey,
+            S.TagValue,
+            @studyKey,
+            (CASE WHEN S.TagLevel <> 2 THEN @seriesKey ELSE NULL END),
```

Comment:
><> [](start = 34, length = 2)

Personally, I'd prefer a more consistent comparison - perhaps if tagLevel < 2 and then if tagLevel < 1?

Reply:
I think both should be fine. There are slightly different on the meaning:
* <>2 => TagLevel is not Study
* <2 => TagLevel is any level that below than Study.